JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/favourites/FavouriteScreenViewModel.kt

```diff
@@ -16,33 +16,99 @@
 
 package com.google.jetstream.presentation.screens.favourites
 
+import androidx.compose.runtime.Immutable
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.google.jetstream.data.entities.MovieList
 import com.google.jetstream.data.repositories.MovieRepository
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.SharingStarted
-import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class FavouriteScreenViewModel @Inject constructor(
     movieRepository: MovieRepository
 ) : ViewModel() {
 
-    val uiState = movieRepository.getFavouriteMovies().map {
-        FavouriteScreenUiState.Ready(it)
+    private val selectedFilterListFlow: MutableSharedFlow<FilterList> = MutableSharedFlow()
+
+    private val selectedFilterList = selectedFilterListFlow.stateIn(
+        viewModelScope,
+        SharingStarted.WhileSubscribed(5_000),
+        FilterList()
+    )
+
+    val uiState = combine(
+        selectedFilterList,
+        movieRepository.getFavouriteMovies()
+    ) { filterList, movieList ->
+        val condition = filterList.toFilterCondition()
+        val filtered = movieList.filterIndexed { index, _ ->
+            condition.idList.contains(index)
+        }
+        FavouriteScreenUiState.Ready(MovieList(filtered), filterList)
     }.stateIn(
         scope = viewModelScope,
         started = SharingStarted.WhileSubscribed(5_000),
         initialValue = FavouriteScreenUiState.Loading
     )
-}
 
+    fun updateSelectedFilterList(filterList: FilterList) {
+        viewModelScope.launch {
+            selectedFilterListFlow.emit(filterList)
+        }
+    }
+
+    companion object {
+        val filterList = FilterList(
+            listOf(
+                FilterCondition.Movies,
+                FilterCondition.TvShows,
+                FilterCondition.AddedLastWeek,
+                FilterCondition.AvailableIn4K
+            )
+        )
+    }
+}
 
 sealed interface FavouriteScreenUiState {
     object Loading : FavouriteScreenUiState
-    data class Ready(val favouriteMovieList: MovieList)
+    data class Ready(val favouriteMovieList: MovieList, val selectedFilterList: FilterList)
```

Comment:
you're missing that this class implements `FavouriteScreenUiState`, which causes the `uiState` to by `StateFlow<Any>`

Reply:
Thanks for catching this. Updating as commented.